File: fb.pl
#!/usr/bin/perl
use File::Slurp;
use strict;
use warnings;
my ($w,$h) = map int, split ",", read_file("/sys/class/graphics/fb0/virtual_size");
print "P3
$w $h
255
";
sub pr { my ($r, $g, $b, $a)=@_; print "$r $g $b\n"; }
my $sq=int($h/2);
my $mc=255;
open(my $fb, ">:raw", "/dev/fb0");
for my $x (0..$h-1) {
    for my $y (0..$w-1) {
        my ($r, $g, $b, $a) = (
            (!(int($x/$sq)%2) | !(int($y/$sq)%2)) ? 0: ($y^$x)% $mc,
            (int($y/$sq) % 2) ? 0 : int($y*$mc/$sq) % $mc,
            (int($x/$sq) % 2) ? 0 : int($x*$mc/$sq) % $mc,
            0,
        );
        pr ($r, $g, $b, $a);
        print $fb pack "C4", ($b, $g,$r ,$a);
    }
}
close($fb)

File: fb.py
#!/usr/bin/env python3
from struct import pack
w,h =map(int, open("/sys/class/graphics/fb0/virtual_size").read().split(","))
midx = w//2
midy = h//2

encode = lambda b,g,r,a : pack("4B",b,g,r,a)

with open("/dev/fb0", "wb") as f:
    for y in range(0,h):
        for x in range(0,w):
            f.write(encode(
                not x%100 and 0xA0 or x<midx and 0xFF or 0, #blue
                y<midy and 0xFF or 0,                       #green
                x>midx and y>midy and 0xFF or 0,            #red
                0,
            ))
